### DISCUSSION TITLE
Reply, resolve, won't fix review comments

### DISCUSSION THREAD

**Discussion by deepakdaniel** [community/community#16514] [Pull Requests]
Something along the lines of this -- https://docs.microsoft.com/en-us/azure/devops/repos/git/review-pull-requests?view=azure-devops&tabs=browser#reply-and-resolve-comments

Not all review comments will be valid. There should be a way to make these review comments at least as
1. accepted and fixed - by the assignee to fix
3. won't fix - does need a fox but not in this PR
4. invalid - the review comment does not apply
The reviewer should be able to resolve or reopen the comments and assignee shouldn't. (A collapse button can be used to replace current resolve button)

In case the review comment is simply a comment i.e. for appreciating a change or something along those lines, it should be flagged as not a review comment.

Why bother with this? There should be a way to glean meaningful metrics from review comments to see that we are doing well in terms of reviews. It would also be helpful in incentivising good peer reviews.

This could probably be done with thumbs up and down emojis today but this simply will not scale. The feature should force PR owners to address review comments in one way or another. i.e. PR comments should all be in one of the mentioned states to be eligible for merge.

**Comment by PmegaOcalc** [comment]
just comment "invalid" and press resolve

**Reply by deepakdaniel** [reply]
Yes, but for folks looking to have some metrics around how PR reviews go on, my feature request would provide a better solution.
Also, review comments would be more easy to action and track for the PR author in case of large PRs (I know, the problem to solve in that case is the size of the PR; still, not always possible!)

**Comment by gtirloni** [comment]
@deepakdaniel what are you trying to solve with this? Developer ego?

**Reply by deepakdaniel** [reply]
Ouch!
But ego aside (though, it was not my focus), there can be real use-cases for this. I'd argue that this will lead to quantifiable measure of PR review quality -- not a perfect measure by any means; but something more reliable than simply the count of comments.
Not every person, team, or org does the best that we can out of PR review. This will just be a small push in hopefully the right direction.
